src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -203,4 +203,62 @@ public bool TryCopyTo(Span<T> destination)
             return true;
         }
     }
+
+    public static class ReadOnlySpanExtensions
+    {
+        /// <summary>
+        /// Creates a new readonly span over the portion of the target string.
+        /// </summary>
+        /// <param name="text">The target string.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown when <paramref name="text"/> is a null
+        /// reference (Nothing in Visual Basic).</exception>
+        public static ReadOnlySpan<char> Slice(this string text)
+        {
+            if (text == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
+
+            return new ReadOnlySpan<char>(ref JitHelpers.GetStringData(text), text.Length);
```

Comment:
You should not need a new JitHelpers method. Adding a new method like `ref char GetFirstCharRef() { return ref m_firstChar; }` should be enough.